cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/cloudstorage/coordinated/CoordinatedCloudStorageDataTransferApi.java

```diff
@@ -0,0 +1,236 @@
+    {
+        forEachInternal(clusterIdFilter, (clusterId, dataTransferApi) -> {
+            restoreJobProgressInternal(clusterId, dataTransferApi, fetchPolicy, progressHandler);
+        });
+    }
+
+    @Nullable
+    @Override
+    public CloudStorageDataTransferApiImpl getValueOrNull(@NotNull String clusterId)
+    {
+        return dataTransferApis.getValueOrNull(clusterId);
+    }
+
+    @Override
+    public void createRestoreJob(CreateRestoreJobRequestPayload createRestoreJobRequestPayload) throws SidecarApiCallException
+    {
+        throw new UnsupportedOperationException("Not supported for coordinated write");
+    }
+
+    @Override
+    public void createRestoreSliceFromExecutor(SidecarInstance sidecarInstance,
+                                               CreateSliceRequestPayload createSliceRequestPayload) throws SidecarApiCallException
+    {
+        throw new UnsupportedOperationException("Not supported for coordinated write");
+    }
+
+    @Override
+    public CompletableFuture<Void> createRestoreSliceFromDriver(SidecarInstance sidecarInstance,
+                                                                CreateSliceRequestPayload createSliceRequestPayload)
+    {
+        throw new UnsupportedOperationException("Not supported for coordinated write");
+    }
+
+    private void createRestoreSliceInternal(CloudStorageDataTransferApiImpl dataTransferApi,
+                                            CreateSliceRequestPayload createSliceRequestPayload) throws SidecarApiCallException
+    {
+        JobInfo jobInfo = dataTransferApi.jobInfo();
+        SidecarClient sidecarClient = dataTransferApi.sidecarClient();
+        QualifiedTableName qualifiedTableName = jobInfo.qualifiedTableName();
+        CreateRestoreJobSliceRequest request = new CreateRestoreJobSliceRequest(qualifiedTableName.keyspace(),
+                                                                                qualifiedTableName.table(),
+                                                                                jobInfo.getRestoreJobId(),
+                                                                                createSliceRequestPayload);
+        RetryPolicy retryPolicy = new CloudStorageDataTransferApiImpl.ExecutorCreateSliceRetryPolicy(sidecarClient);
+        RequestContext requestContext = sidecarClient.requestBuilder().retryPolicy(retryPolicy).request(request).build();
+        try
+        {
+            sidecarClient.executeRequestAsync(requestContext).get();
+        }
+        catch (Exception exception)
+        {
+            handleInterruption(exception);
+            throw new SidecarApiCallException("Failed to create restore job slice. " +
+                                              "restoreJobId=" + jobInfo.getRestoreJobId() +
+                                              " sliceId=" + createSliceRequestPayload.sliceId(),
+                                              exception);
+        }
```

Comment:
This method, minus the extra exception handling, is almost identical to the one in `CloudStorageDataTransferApiImpl` - could you not find a way to simply call that one from here, maybe adding the exception handling there so you could just call that one?